jena-arq/src/main/java/org/apache/jena/atlas/data/SortedDataBag.java

```diff
@@ -69,19 +69,81 @@
     
     protected final ThresholdPolicy<E> policy;
     protected final SerializationFactory<E> serializationFactory;
-    protected final Comparator<? super E> comparator;
+    protected final CanAbortComparator comparator;
     
     protected boolean finishedAdding = false;
     protected boolean spilled = false;
     protected boolean closed = false;
+    protected volatile boolean cancelled;
```

Comment:
By making this a field inside `CanAbortComparator` and offering a method `cancel()` on `CanAbortComparator` you could break that type out and keep it static, instead of an inner type here.